File: Crio/crio_programming_interview_problems-master/MergeSortedArray/MergeSortedArray.java
import java.io.*;
import java.util.*;

public class MergeSortedArray {
    // Implement your solution by completing the below function
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int p1 = m - 1;
        int p2 = n - 1;
        int p = m + n - 1;

        while ((p1 >= 0) && (p2 >= 0))
            nums1[p--] = (nums1[p1] < nums2[p2]) ? nums2[p2--] : nums1[p1--];

        System.arraycopy(nums2, 0, nums1, 0, p2 + 1);
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int m = scanner.nextInt();
        int n = scanner.nextInt();
        int[] nums1 = new int[m + n];
        int[] nums2 = new int[n];

        for (int i = 0; i < m; i++)
            nums1[i] = scanner.nextInt();
        for (int i = 0; i < n; i++)
            nums2[i] = scanner.nextInt();

        scanner.close();

        new MergeSortedArray().merge(nums1, m, nums2, n);
        for (int i = 0; i < nums1.length; i++)
            System.out.print(nums1[i] + " ");
    }
}

File: Crio/crio_programming_interview_problems-master/MergeSortedArray/MergeSortedArray.cpp
#include <bits/stdc++.h>
#include "FastIO.hpp"
#include "ReadMatrix.hpp"
#include "PrintMatrix.hpp"

using namespace std;

class MergeSortedArray {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
    	// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int x = nums1.size();
        if(m==1 && x==1) return;
        int p1 = 0, p2 = 0;
        while(p1 < x && p2 < n) {
            if((x-p1) == (n-p2)) {
                nums1[p1++] = nums2[p2++];
            } else if(nums1[p1] <= nums2[p2]) {
                p1++;
            } else {
                for(int i = x - 1; i > p1; i--) {
                    nums1[i] = nums1[i-1];
                }
                nums1[p1] = nums2[p2];
                p2++;
            }
        }
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }
};

int main() {
	FastIO();
	int m, n;
	cin >> m >> n;
	vector<int> nums1, nums2;
	ReadMatrix<int>().OneDMatrix(m, nums1);
	ReadMatrix<int>().OneDMatrix(n, nums2);
	nums1.resize(m + n);
	MergeSortedArray().merge(nums1, m, nums2, n);
	PrintMatrix<int>().OneDMatrix(nums1, " ");
	return 0;
}
